main.py

```diff
@@ -1,4 +1,27 @@
+import threading
+import pigpio
+
 from api import app
+from camera import Camera
+from pid import PID
+from servo import Servo
+from balancer import vision
+
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=5000, debug=False)
+    pi = pigpio.pi()
```

Comment:
you could use `from pigpio import pi` and write `_pi = pi()`, so you don't import the entire library for a function

Reply:
same for `threading`